packages/frontend/app/components/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,161 @@
+                  enterFrom='opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95'
+                  enterTo='opacity-100 translate-y-0 sm:scale-100'
+                  leave='ease-in duration-200'
+                  leaveFrom='opacity-100 translate-y-0 sm:scale-100'
+                  leaveTo='opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95'
+                >
+                  <Dialog.Panel className='relative transform overflow-hidden rounded-lg max-w-full transition-all bg-white px-4 pb-4 pt-5 text-left shadow-xl w-full sm:max-w-lg'>
+                    <div className='absolute right-0 top-0 pr-4 pt-4'>
+                      <button
+                        type='button'
+                        className='text-gray-400 hover:text-gray-500 focus:outline-none'
+                        onClick={() => cancelHandler()}
+                      >
+                        <span className='sr-only'>Close</span>
+                        <X className='h-8 w-8' aria-hidden='true' />
+                      </button>
+                    </div>
+                    <div>
+                      <div className='mx-auto flex h-12 w-12 items-center justify-center rounded-full bg-red-100'>
+                        <ExclamationTriangle
+                          className='text-red-500 h-5 w-5'
+                          aria-hidden='true'
+                        />
+                      </div>
+                      <div className='mt-3 text-center'>
+                        <Dialog.Title
+                          as='h3'
+                          className='font-semibold leading-6 text-lg'
+                        >
+                          {title}
+                        </Dialog.Title>
+                        <div className='mt-2'>
+                          <p>
+                            {message
+                              ? message
+                              : `Please note that this action is not reversible.`}
+                          </p>
+                          <p className='font-medium mt-4'>
+                            To confirm, type &quot;
+                            <span className='font-bold'>{keyword}</span>
+                            &quot; below:
+                          </p>
+                          <Input
+                            value={confirmationPrompt}
+                            ref={inputRef}
+                            onChange={(e) =>
+                              setConfirmationPrompt(e.currentTarget.value)
+                            }
+                          />
+                        </div>
+                        <div className='mt-2'>
+                          <Button
+                            aria-label={confirmButtonText ?? 'Confirm'}
+                            intent='danger'
+                            onClick={() => confirmHandler()}
+                            className='w-full'
+                            disabled={!(confirmationPrompt === keyword)}
```

Comment:
```suggestion
                            disabled={confirmationPrompt.toLowerCase() !== keyword.toLowerCase()}
```

Reply:
and maybe we can pull this into a method so we can reuse it in L43

Reply:
It does make sense to extract this into a method. However, I believe transforming the string to lowercase would not be beneficial in this case. If we make `peer.name` a required field and the peer's name is 'My first PEER,' I think it would be reasonable to expect the user to input the exact peer name without case modifications.

Reply:
I was just following how github archiving/deleting repository UX worked and it was indeed case insensitive, I don't feel strongly about it